Functional.Primitives/ResultFactory.cs

```diff
@@ -175,6 +175,38 @@ public static class Result
 			}
 		}
 
+		public static Result<TSuccess, TException> Try<TSuccess, TException>(Func<TSuccess> successFactory) where TException : Exception
+		{
```

Comment:
Is it possible to DRY out the other `Try` methods by calling these ones?

Reply:
Yea, should be pretty straightforward. Means @RyanMarcotte will have to review and approve yet again though ;)

Reply:
Well, I DRY'd up the two Try methods, but wasn't able to figure out a way to write `Try<Unit, TException>()` as a call to `Try<TSuccess, TException>()` without introducing a new lambda (which failed the "No Object Allocations" test). Any thoughts on that one?

Reply:
Ahh yes, that would cause issues there wouldn't it. Slightly DRYer is always better :)

Reply:
The code is simple enough that I'm not too worried about the duplication to be honest.  Depends on how much we want to avoid the allocation really.

Reply:
> Slightly DRYer is always better :)

That is a bold claim @NathanMagnus :)
https://blog.ploeh.dk/2014/08/07/why-dry/

>It's not a new observation that unconditional elimination of duplicated code can be harmful. The Rule of Three exists for this reason:
>- Write a piece of code.
>- Write the same piece of code again. Resist the urge to generalise.
>- Write the same piece of code again. Now you are allowed to consider generalising it.

YMMV, but I've found the above heuristic works well in practice.  I've just included it here for the benefit of more junior developers.

Reply:
Big +1 on the "it depends" for DRYer is always better, because the consequence of DRY is coupling. In the local scope of a single class, that coupling isn't really a problem, but in larger scopes...